# Exercise 1

In [1]:
import spacy
from string import punctuation
from spacy.lang.en.stop_words import STOP_WORDS

nlp=spacy.load('en_core_web_sm')

file_name = './data/spaceX_DP.txt'

with open(file_name, 'r') as reader:
    spacex_dp = reader.read()

spacex_dp_x = nlp(spacex_dp)
print(spacex_dp_x)

Cape Canaveral this is not.But here, down toward the coast, on a spit of land past the Border Patrol checkpoint, where the Rio Grande meets the Gulf of Mexico, there is a spaceship being assembled off State Highway 4 just before it dead-ends into the sea.Towering and stainless-steel shiny, it looks like a surreal sculpture amid the cactuses, yucca and relentless South Texas sun. And, because it’s being built not in a factory but out in the open, it’s become a roadside attraction, drawing gawkers to an area so remote that the county trucks in drinking water once a month to the few who live nearby.They’re coming to see Elon Musk’s latest creation, a prototype called Starship that he hopes will one day carry people by the dozens to the moon and Mars. Musk, in a presentation here Saturday, said his goal of building a “rapidly reusable spacecraft” here would lead to the fulfillment of his ultimate goal of creating “a city on Mars.” But first, he’ll need to pull off another improbable feat, 

Word frequency and removing stopwords

In [2]:
stop_words = list(STOP_WORDS)
spacex_dp_tokens = [token.text for token in spacex_dp_x]
freq_word={}
maximum_frequency = 0
for w in spacex_dp_tokens:
    w_lower = w.lower()

    if w_lower not in stop_words and w_lower.isalpha():
        if w_lower in freq_word.keys():
            freq_word[w_lower] += 1
        else:
            freq_word[w_lower] = 1
            
        if maximum_frequency < freq_word[w_lower]:
            maximum_frequency = freq_word[w_lower]
            
freq_word

{'cape': 9,
 'canaveral': 7,
 'coast': 7,
 'spit': 2,
 'land': 15,
 'past': 8,
 'border': 29,
 'patrol': 3,
 'checkpoint': 5,
 'rio': 9,
 'grande': 9,
 'meets': 1,
 'gulf': 6,
 'mexico': 7,
 'spaceship': 8,
 'assembled': 4,
 'state': 24,
 'highway': 19,
 'dead': 2,
 'ends': 1,
 'sea': 12,
 'towering': 2,
 'stainless': 6,
 'steel': 9,
 'shiny': 1,
 'looks': 3,
 'like': 38,
 'surreal': 2,
 'sculpture': 1,
 'amid': 2,
 'cactuses': 1,
 'yucca': 2,
 'relentless': 1,
 'south': 41,
 'texas': 76,
 'sun': 5,
 'built': 11,
 'factory': 2,
 'open': 6,
 'roadside': 1,
 'attraction': 2,
 'drawing': 1,
 'gawkers': 1,
 'area': 28,
 'remote': 2,
 'county': 37,
 'trucks': 7,
 'drinking': 2,
 'water': 14,
 'month': 17,
 'live': 9,
 'nearby': 12,
 'coming': 1,
 'elon': 19,
 'musk': 85,
 'latest': 5,
 'creation': 1,
 'prototype': 23,
 'called': 13,
 'starship': 67,
 'hopes': 2,
 'day': 24,
 'carry': 5,
 'people': 32,
 'dozens': 2,
 'moon': 12,
 'mars': 26,
 'presentation': 5,
 'saturday': 5,
 'said': 127,


Maximum frequency

In [3]:
word_frequencies = {}

for word in freq_word.keys():
    word_frequencies[word] = (freq_word[word]/maximum_frequency)
    
word_frequencies

{'cape': 0.03543307086614173,
 'canaveral': 0.027559055118110236,
 'coast': 0.027559055118110236,
 'spit': 0.007874015748031496,
 'land': 0.05905511811023622,
 'past': 0.031496062992125984,
 'border': 0.1141732283464567,
 'patrol': 0.011811023622047244,
 'checkpoint': 0.01968503937007874,
 'rio': 0.03543307086614173,
 'grande': 0.03543307086614173,
 'meets': 0.003937007874015748,
 'gulf': 0.023622047244094488,
 'mexico': 0.027559055118110236,
 'spaceship': 0.031496062992125984,
 'assembled': 0.015748031496062992,
 'state': 0.09448818897637795,
 'highway': 0.07480314960629922,
 'dead': 0.007874015748031496,
 'ends': 0.003937007874015748,
 'sea': 0.047244094488188976,
 'towering': 0.007874015748031496,
 'stainless': 0.023622047244094488,
 'steel': 0.03543307086614173,
 'shiny': 0.003937007874015748,
 'looks': 0.011811023622047244,
 'like': 0.14960629921259844,
 'surreal': 0.007874015748031496,
 'sculpture': 0.003937007874015748,
 'amid': 0.007874015748031496,
 'cactuses': 0.0039370078740

Sentence score

In [4]:
sentence_score = {}

for sent in spacex_dp_x.sents:
    for word in sent:
        if word.text.lower() in freq_word.keys():
            if sent in sentence_score.keys():
                sentence_score[sent] += freq_word[word.text.lower()]
            else:
                sentence_score[sent] = freq_word[word.text.lower()]
                
sentence_score

{Cape Canaveral this is not.: 16,
 But here, down toward the coast, on a spit of land past the Border Patrol checkpoint, where the Rio Grande meets the Gulf of Mexico, there is a spaceship being assembled off State Highway 4 just before it dead-ends into the sea.: 171,
 Towering and stainless-steel shiny, it looks like a surreal sculpture amid the cactuses, yucca and relentless South Texas sun.: 190,
 And, because it’s being built not in a factory but out in the open, it’s become a roadside attraction, drawing gawkers to an area so remote that the county trucks in drinking water once a month to the few who live nearby.: 152,
 They’re coming to see Elon Musk’s latest creation, a prototype called Starship that he hopes will one day carry people by the dozens to the moon and Mars.: 317,
 Musk, in a presentation here Saturday, said his goal of building a “rapidly reusable spacecraft” here would lead to the fulfillment of his ultimate goal of creating “a city on Mars.”: 304,
 But first, he’

Summarized sentences

In [5]:
from heapq import nlargest

top30percent_sentence=int(0.3*len(sentence_score))
summarized_sentences = nlargest(top30percent_sentence, sentence_score)
print(summarized_sentences)

[“We’ll do several short hops to smooth out launch process, then go high altitude with body flaps,” Musk tweeted., SpaceX hasn’t disclosed details about upcoming tests, although Federal Aviation Administration temporary flight restrictions to support hop tests remain in place through Aug. 6., We’re going to send people to Mars and make life multiplanetary,” he said., We’re going to have a base on the moon., “We’re going to go the moon., Musk was at the same event, calling the successful Demo-2 flight the beginning of a “new era in space exploration” that included going to the moon and Mars., “You have delivered beyond anything any of us would have expected.”, “I want to tell you, Elon, you responded absolutely magnificently,” Bridenstine said at an Aug. 2 ceremony at Ellington Field, near the Johnson Space Center, to welcome back the astronauts., Bridenstine said he was pleased, recalling those earlier comments., ”SpaceX did deliver with the Demo-2 mission, which successfully flew two 

Summary string

In [6]:
summary_text_sents = []

for sent in summarized_sentences:
    summary_text_sents.append(sent.text)

summary = ' '.join(summary_text_sents)
print(f'Summary length: {len(summary)}')
print(f'Original length: {len(spacex_dp)}')
print(summary)

Summary length: 43824
Original length: 121221
“We’ll do several short hops to smooth out launch process, then go high altitude with body flaps,” Musk tweeted. SpaceX hasn’t disclosed details about upcoming tests, although Federal Aviation Administration temporary flight restrictions to support hop tests remain in place through Aug. 6. We’re going to send people to Mars and make life multiplanetary,” he said. We’re going to have a base on the moon. “We’re going to go the moon. Musk was at the same event, calling the successful Demo-2 flight the beginning of a “new era in space exploration” that included going to the moon and Mars. “You have delivered beyond anything any of us would have expected.” “I want to tell you, Elon, you responded absolutely magnificently,” Bridenstine said at an Aug. 2 ceremony at Ellington Field, near the Johnson Space Center, to welcome back the astronauts. Bridenstine said he was pleased, recalling those earlier comments. ”SpaceX did deliver with the Demo-2 m

Comparison with gensim summary

In [7]:
from gensim.summarization import summarize

print ('Summary:')
print (summarize(spacex_dp, word_count=250))

Summary:
Musk, in a presentation here Saturday, said his goal of building a “rapidly reusable spacecraft” here would lead to the fulfillment of his ultimate goal of creating “a city on Mars.” But first, he’ll need to pull off another improbable feat, building a private, commercial spaceport here, in what the top local elected official called a “mind-boggling” juxtaposition: SpaceX, one of the hottest companies in the world, led by a Silicon Valley celebrity with nearly 29 million Twitter followers, building a rocket in a border town where nearly a third of the residents live below the poverty line.“I never in a million years would have imagined it,” said Cameron County Judge Eddie Treviño Jr. Five years ago, SpaceX started building a launchpad here, hauling in dirt by the ton, that would allow the company a measure of freedom without the restraints that come with shooting rockets off from government sites, such as Cape Canaveral or Vandenberg Air Force Base in California, where several

# Exercise 2

In [13]:
import nltk
from nltk.sentiment.vader import SentimentIntensityAnalyzer
nltk.download('vader_lexicon')
from textblob import TextBlob

[nltk_data] Downloading package vader_lexicon to
[nltk_data]     /home/jorgenv/nltk_data...


True

In [15]:
sid = SentimentIntensityAnalyzer()
polarity_score = sid.polarity_scores(spacex_dp)

testimonial = TextBlob(spacex_dp)
print(polarity_score)
print(testimonial.sentiment)

{'neg': 0.048, 'neu': 0.874, 'pos': 0.078, 'compound': 1.0}
Sentiment(polarity=0.07758178365698393, subjectivity=0.4410466259483019)


The VADER sentiment analysis categorizes most of the words as neutral (87%), but has a positive compound score, which means the general sentiment of the text is positive.

The TextBlob analysis categorizes the text as very neutral sentiment (neither positive or negative), while being slightly more objective in nature.